localstack/services/cloudformation/scaffolding/__main__.py

```diff
@@ -0,0 +1,345 @@
+            if "type" in defn and defn["type"] in ["string"]:
+                yield name
+
+    def render_template(self, resource_name: ResourceName) -> str:
+        template = {
+            "AWSTemplateFormatVersion": "2010-09-09",
+            "Description": f"Template to exercise {resource_name.full_name}",
+            "Parameters": {
+                "AttributeName": {
+                    "Type": "String",
+                    "Description": "Name of the attribute to fetch from the resource",
+                },
+            },
+            "Resources": {
+                "MyResource": {
+                    "Type": resource_name.full_name,
+                    "Properties": {},
+                },
+            },
+            "Outputs": self.render_outputs(),
+        }
+
+        return safe_dump(template, sort_keys=False)
+
+    def required_properties(self) -> dict[str, Property]:
+        return PropertyRenderer(self.schema).properties()
+
+    def render_outputs(self) -> dict:
+        """
+        Generate an output for each property in the schema
+        """
+        outputs = {}
+
+        # ref
+        outputs["MyRef"] = {"Value": {"Ref": "MyResource"}}
+
+        # getatt
+        outputs["MyOutput"] = {"Value": {"Fn::GetAtt": ["MyResource", {"Ref": "AttributeName"}]}}
+
+        return outputs
+
+
+class PropertyRenderer:
+    def __init__(self, schema: ResourceSchema):
+        self.schema = schema
+
+    def properties(self) -> dict:
+        required_properties = self.schema.get("required", [])
+
+        result = {}
+        for name, defn in self.schema["properties"].items():
+            if name not in required_properties:
+                continue
+
+            value = self.render_property(defn)
+            result[name] = value
+
+        return result
+
+    def render_property(self, property: Property) -> str | dict | list:
+        if prop_type := property.get("type"):
+            if prop_type in {"string"}:
+                return self._render_basic(prop_type)
+            elif prop_type == "array":
+                return [self.render_property(item) for item in property["items"]]
+        elif oneof := property.get("oneOf"):
+            return self._render_one_of(oneof)
+        else:
+            raise NotImplementedError(property)
+
+    def _render_basic(self, type: str) -> str:
+        return "CHANGEME"
+
+    def _render_one_of(self, options: list[Property]) -> Any:
+        return self.render_property(options[0])
+
+
+class FileWriter:
+    def __init__(self, root: Path, resource_name: ResourceName, console: Console):
+        self.root = root
+        self.resource_name = resource_name
+        self.console = console
+
+    def write_provider(self, contents: str):
+        destination = self.root.joinpath(
+            "localstack",
+            "services",
+            self.resource_name.service.lower(),
+            "resource_providers",
+            f"{self.resource_name.resource.lower()}.py",
```

Comment:
I think I'd prefer a name with the full resource type encoded, e.g. `aws_opensearch_domain.py` here. 
wdyt?

Reply:
:shrug: I'm not bothered.

Reply:
:shrug: I'm not bothered.

Reply:
:shrug: I'm not bothered.